File: Algotithm_Implement/Bubble_sort/Bubblesort.java
import java.util.*;
import java.io.*;

class Bubblesort {
    static int [] arr = {2, 5, 3, 1, 4};

    public static void main(String[] args) {
        bubbleSort();

        for(int a : arr) {
            System.out.println(a);
        }
    }

    public static void bubbleSort() {
        for(int i = 0; i < arr.length; i++) {
            for(int j = 1; j < arr.length - i; j++) {
                if(arr[j - 1] > arr[j]) {
                    int temp = arr[j - 1];
                    arr[j - 1] = arr[j];
                    arr[j] = temp;
                }
            }
        }
    }
}

File: Algotithm_Implement/Sort/Bubble_sort/Bubble_sort.cc
#include <bits/stdc++.h>

using namespace std;

vector<int> arr;

void Print() {
    for (int a : arr) {
        cout << a << " ";
    }
    cout << "\n";
}

void bubbleSort() {
    clock_t start, finish;
    double duration;

    start = clock();

    for (int i = 0; i < arr.size() - 1; i++) {
        bool isSwap = false;
        // 뒤에서 i개는 정렬이 무조건 되기 때문에 i씩 빼주기
        for (int j = 0; j < arr.size() - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
                isSwap = true;
            }
        }
        if (!isSwap)
            break;
    }

    finish = clock();

    duration = (double)(finish - start) / CLOCKS_PER_SEC;
    Print();
    cout << duration << "초" << "\n";
}

int main() {
    // int n;
    // cin >> n;

    for (int i = 0; i < 2500; i++) {
        // int a;
        // cin >> a;
        arr.push_back(rand() % 100);
    }

    bubbleSort();
}